synapse/storage/_base.py

```diff
@@ -409,16 +409,16 @@ def _new_transaction(
             i = 0
             N = 5
             while True:
+                cursor = conn.cursor()
+                cursor = LoggingTransaction(
+                    cursor,
```

Comment:
can we call `conn.cursor()` in here instead of redefining a variable? this will bite us when mypy gets to it eventually :)